index.js

```diff
@@ -146,6 +147,13 @@ function session(options){
     if (cookieOptions.secure === 'auto') {
       req.session.cookie.secure = issecure(req, trustProxy);
     }
+
+    if (cookieOptions.autoSubDomain) {
+      var parsedDomain = parseDomain(req.headers.host);
```

Comment:
Instead of only using `req.headers.host` directly, we need to actually hook into the mechanisms of Express.js such that it uses the settings and parsing that is in there so people who use "trust proxy" and the like can have the correct domain when using a reverse proxy.